app/src/main/java/com/github/gotify/service/WebSocketService.java

```diff
@@ -323,6 +323,30 @@ private void showNotification(
                 .setColor(ContextCompat.getColor(getApplicationContext(), R.color.colorPrimary))
                 .setContentIntent(contentIntent);
 
+        String actionOpen =
```

Comment:
What are your use-cases for both features, and why do we need another url open, when there is already https://gotify.net/docs/msgextras#clientnotification click.url

Reply:
> This is to be able to open another url for example from the actions buttons under the notification

The current implementation seems too unflexible. You could already have functionality in mark down for opening a URL. Maybe we could parse the links inside a markdown message and automatically create buttons for these. Otherwise, a more generic approach would look like:
```json5
{
  "extras": {
    "client::action": {
      "buttons": [
        {
          "label": "Acknowledge",
          "http": {
            "method": "post",
            "url": "https://example.com/ack?id=5",
            "headers": {
              "Authorization": "bearer abcde"
            }
          }
        },
        // another button
      ]
    }
  }
}
```
But something like this would require more input from other users, so that we know this is fulfills their use cases. Perhaps there is already a standard for defining actions like this, f.ex Microsoft Teams supports such buttons.

Could you give me a concrete use-case for the share feature.